### PR TITLE
refactor(app): Refactor display name utils

### PR DESCRIPTION
**Pull request by mjhuff**
Closes [EXEC-773](https://opentrons.atlassian.net/browse/EXEC-773)

# Overview

Now that we have a definitive locations for general app utils, all general utilities should now be imported through places like `local-resources` and not from the internals of component directories. One of the biggest culprits of violating this rule is `Commands`. We use a lot of these utilities elsewhere. 

This PR serves as a general refactor for display name utils, which entails:

- Migrating them to `local-resources`
- Tightening the interfaces. More specifically, instead of passing all of `protocolAnalysis`, pass only what we need.
- The above has the added benefit of highlighting potential problem parameters, ie, `commands`. I actually caught a couple more unnecessary cases of iterating through `protocolAnalysis`, so this refactor reduces TC by O(2n) in some spots.
- The above also enforces stricter typing - there are actually a couple spots that could cause whitescreens on older protocols, because we weren't properly typechecking protocolAnalysis before attempting to iterate through `modules` or `labware` when it could in fact be an object and not an array. I guess in practice, this probably doesn't happen much, since I've personally never seen any tickets about this.

[04b6ad3](https://github.com/Opentrons/opentrons/pull/16537/commits/04b6ad3f6f9b96ec7e5996fd9d3bcfbee5896d6d) Contains the functional changes. 

[d350e0d](https://github.com/Opentrons/opentrons/pull/16537/commits/d350e0d4fd1386220bc94dd7db39a601d7f7e760) is just a million lint/test fixes.

## Test Plan and Hands on Testing
- Smoke tested runs. Everything looks good.

## Risk assessment
lowish-medium. 

[EXEC-773]: https://opentrons.atlassian.net/browse/EXEC-773?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
We know we're in a good place when we're using `Array.isArray`

Good finds on removing more command iteration!

File: app/src/local-resources/labware/utils/getAllDefs.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { getAllDefinitions } from '@opentrons/shared-data'
+
```
Comment (TamarZanzouri) [inline app/src/local-resources/labware/utils/getAllDefs.ts:2]: added by mistake?

**Review by TamarZanzouri — APPROVED** [review]
WOW! so much better! 1 unintentional change but besides that looks great

File: app/src/local-resources/labware/utils/getAllDefs.ts (hunk 1 above)
Comment (mjhuff) [reply app/src/local-resources/labware/utils/getAllDefs.ts:2]: No, it's intentional.

This is documented somewhere on confluence, but generally we have a pattern in the monorepo for js import order:

1. Global imports
2. Top-level monorepo imports
3. Local imports
4. Types

These should be separated by a space.

**Pull request merged**